packages/itwin/infrastructure-iot/src/components/partials/inputs/InputDatePicker.tsx

```diff
@@ -0,0 +1,55 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+
+import React, { useEffect, useState } from "react";
+
+import { DatePicker, DropdownButton, MenuExtraContent, useTheme } from "@itwin/itwinui-react";
+
+import { UtilitiesService } from "../../../services/UtilitiesService";
+
+import styles from "./InputDatePicker.module.scss";
+
+export function InputDatePicker(props: {value: string, onChange: (newValue: string) => void}) {
+
+  // This is needed to iTwin React component to respect system-wide light/dark mode
+  useTheme("os");
+
+  // Storage for date object of input
+  const [date, setDate] = useState<Date | undefined>(new Date(props.value));
+  useEffect(() => {
```

Comment:
The component does re-render when the props change, but the state will not update to the new value without the `useEffect` hook - it will remain set to whatever it was set to previously. I have tested this and verified that the hook is needed if you want to correctly update the state after a prop change.

I found some literature online supporting this as well.